File: Programmers/스택.큐/disguise.py
def solution(clothes):
    answer = 1
    disguise = dict()
    for i in clothes:
        category = i[1]
        if category in disguise:
            disguise[category] += 1
        else:
            disguise[category] = 1
    for i in disguise.values():
        answer *= (i + 1)
    return answer - 1

print(solution([["yellow_hat", "headgear"], ["blue_sunglasses", "eyewear"], ["green_turban", "headgear"]]))

File: Programmers_Java/해시/Disguise.java
package Programmers;

import java.util.ArrayList;
import java.util.HashMap;

public class Disguise {
    public int solution(String[][] clothes) {
        HashMap<String, Integer> map = new HashMap<>();
        for (String[] cloth : clothes) {
            map.put(cloth[1], map.getOrDefault(cloth[1], 1) + 1);
        }

        int answer = 1;
        for(Integer n : map.values()){
            answer *= n;
        }

        return answer - 1;
    }
}
